corehq/apps/users/models.py

```diff
@@ -1777,6 +1777,7 @@ def unset_location_by_id(self, location_id, fall_back_to_next=False):
     def reset_locations(self, location_ids):
         """
         Reset user's assigned_locations to given location_ids and update user data.
+            This should be called after updating primary location via set_location/unset_location
```

Comment:
It's hard to enforce it, as it's hard to check from the data whether primary location is updated or not. I was considering to just combining resetting locations and setting primary location into just this one method, but there are multiple ways of checking/enforcing consistency on whether a primary location should be set or not. Instead I thought it's better to leave that logic to callers.